net/http/civetweb/civetweb.c

```diff
@@ -2697,6 +2697,11 @@ mg_set_thread_name(const char *name)
 #endif
 #elif defined(_GNU_SOURCE) && defined(__GLIBC__) \
     && ((__GLIBC__ > 2) || ((__GLIBC__ == 2) && (__GLIBC_MINOR__ >= 12)))
+    #if defined(__clang__)
+    /* Show no warning in case implicit declaration of function 'pthread_setname_np'. */
+    #pragma clang diagnostic push
+    #pragma clang diagnostic ignored "-Wimplicit-function-declaration"
+    #endif
```

Comment:
This only suppress warning when compiled with clang. 
But does it works? Is anybody tried to start THttpServer after compilation?

`pthread_setname_np` function is extra functionality, which is provided by GLIBC on Linux.
If I understood correctly, problem appears on Mac, therefore one should test it on Mac.

I found similar problem here:  https://github.com/Pithikos/C-Thread-Pool/issues/55
And proposed solution is different.

Reply:
As I said before, this doesn't make much sense to me.

The warning says that we have no declaration of this function. We should `#include` the relevant header with the relevant preprocessor definitions to make the declaration visible (instead of suppressing the warning).

Reply:
@Axel-Naumann 
In mentioned link there is solution - but it uses different way as civetweb does.
I will try different patch myself and then let you know

Reply:
There is alternative solution: https://github.com/root-project/root/pull/1966
One should always define `_GNU_SOURCE`. 
@oshadura can you test it in your environment?

Reply:
Thanks! I will let you know